arm/Microsoft.CognitiveServices/accounts/deploy.bicep

```
@@ -209,7 +203,11 @@ resource cognitiveServices 'Microsoft.CognitiveServices/accounts@2021-10-01' = {
   }
   properties: {
     customSubDomainName: !empty(customSubDomainName) ? customSubDomainName : null
-    networkAcls: !empty(networkAcls) ? networkAcls_var : null
+    networkAcls: !empty(networkAcls) ? {
+      defaultAction: contains(networkAcls, 'defaultAction') ? networkAcls.defaultAction : 'Deny'
+      virtualNetworkRules: contains(networkAcls, 'virtualNetworkRules') ? networkAcls.virtualNetworkRules : []
+      ipRules: contains(networkAcls, 'ipRules') ? networkAcls.ipRules : []
+    } : null
```

Same as disabling public network access I'd keep the default behavior here
```suggestion
    networkAcls: !empty(networkAcls) ? {
      defaultAction: contains(networkAcls, 'defaultAction') ? networkAcls.defaultAction : null
      virtualNetworkRules: contains(networkAcls, 'virtualNetworkRules') ? networkAcls.virtualNetworkRules : []
      ipRules: contains(networkAcls, 'ipRules') ? networkAcls.ipRules : []
    } : null
```

---

Same comment as above. What takes precidence: Security recommendations, or module usability? I honestly can't tell.

---

I think this is dependent on a wider discussion as per issue #819, similarly to PR #1453. I've put that back to triage and applied the needs further discussion label so that we can align on a common approach for all modules. In general I think we should if possible consider all the 3 components of the CIA triad when dealing with security: Confidentiality, Integrity and Availability. By locking resource access we're not taking Availability into account, which is still part of the Security topic.

---

Agreed. The only thing we make sure is, that if somebody e.g. deploys the module with a private endpoint, they don't forget to lock the service down